### PR TITLE
[maskedtensor] move __torch_function/dispatch__ functions to a map

### PR DESCRIPTION
**Pull request by george-qi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85529

### CODE REVIEW

File: torch/masked/maskedtensor/core.py [hunk 1]
```diff
@@ -383,21 +262,10 @@ def __repr__(self):
...
-            return _MaskedWhere.apply(*args)
-        if func is torch.Tensor.contiguous:
-            return _MaskedContiguous.apply(args[0])
-        if func is torch.Tensor.to_dense:
-            return _MaskedToDense.apply(args[0])
-        if func is torch.Tensor.to_sparse:
-            return _MaskedToSparse.apply(args[0])
-        if func is torch.Tensor.to_sparse_csr:
-            return _MaskedToSparseCsr.apply(args[0])
+        from ._ops_refs import _MASKEDTENSOR_FUNCTION_TABLE
```
Comment (bhosmer) [inline torch/masked/maskedtensor/core.py:265]: I know I'm being a broken record about this, but: it's always better to have imports at the top of a module, unless you're in some rare situation where a) import is very heavy and b) the (single) function that needs it might never be called. see e.g. https://stackoverflow.com/questions/9614051/import-at-module-level-or-at-function-level

File: torch/masked/maskedtensor/core.py [hunk 2]
```diff
@@ -415,161 +283,10 @@ def unary(cls, fn, data, mask):
...
-
-            new_data = func(*new_args, **kwargs)
-            new_args[-1] = torch.ones_like(new_args[-1])
-            new_mask = func(*new_args, **kwargs).bool()
-
-            return MaskedTensor(new_data, new_mask)
-        if func is torch.ops.aten.is_same_size:
-            _check_args_kwargs_length(args, kwargs, f"__torch_dispatch__, {func}", len_args=2)
-            return data.is_same_size(_get_data(args[1]))
+        from ._ops_refs import _MASKEDTENSOR_DISPATCH_TABLE
```
Comment (bhosmer) [inline torch/masked/maskedtensor/core.py:286]: same nit re imports at top

**Review by bhosmer — APPROVED** [review]
nice! exactly what I was thinking 😁 assuming green tests looks GTG to me (mod import nit)

File: torch/masked/maskedtensor/core.py (hunk 1 above)
Comment (george-qi) [reply torch/masked/maskedtensor/core.py:265]: No you're good! This one was necessary because `._ops_refs` imports from `core.py`, so this was necessary because of circular imports

File: torch/masked/maskedtensor/core.py (hunk 2 above)
Comment (george-qi) [reply torch/masked/maskedtensor/core.py:286]: same thing as above

File: torch/masked/maskedtensor/core.py (hunk 1 above)
Comment (bhosmer) [reply torch/masked/maskedtensor/core.py:265]: ah got it, thanks!
